### ISSUE TITLE
Option to not integrate with invoke

### ISSUE THREAD

**Issue by jawadst** [serverless-heaven/serverless-webpack#288]
# This is a Feature Proposal

## Description

It'd be great to have an option to not run webpack when running invoke. Locally, I test with node 8 and only need webpack/babel when deploying to Lambda.

**Comment by HyperBrain** [comment]
Hi @jawadst ,

thanks for opening the feature request. This option would disable the whole plugin on request (e.g. if a `--webpack-disable`) switch would be given. Maybe the functionality would fit better into Serverless - i.e. provide a command line switch there, to disable arbitrary plugins with a command line switch, as it would make sense for a few other plugins too.

However, by disabling webpack on demand, I see some drawbacks:
* The local tests are not representative anymore, because you test different code than the one actually deployed - in fact they are not representative already, as you use a different Node version locally ;-)
* The `invoke local --watch` variant of the command is disabled too, to have a continuous test running after code changes.

A change in Serverless to support a generic switch looks trivial to me. I can implement it and open a PR there that references this issue. Then it would be in Serverless 1.25, or more likely 1.26.

What do you think?

**Comment by jawadst** [comment]
Hi @HyperBrain,

I agree that enabling this option makes the tests less representative. That being said, we have been doing this for a while (we usually develop with node >= 8 and run our tests that way too) without any issue so far so my take on this is that the benefits seem higher than the risk for us. I totally understand some people might disagree on that which is why I was thinking of it as an option.

As for the `invoke local --watch`, my understanding is that your generic switch would allow to either keep the plugin enabled (no switch) or disabled for that command as well, right?

Thanks a lot for your response and your work on this plugin, it's very helpful!

**Comment by HyperBrain** [comment]
Yes, your assumption is right - the plugin would be completely disabled.
I will create an issue in Serverless for the functionality and might be able to implement it soon.

**Comment by jawadst** [comment]
Souds great, thanks you very much!

**Comment by HyperBrain** [comment]
@jawadst This might be partially solved with #341. `invoke local` now supports a `--no-build` option that runs the invoke on already compiled code. Of course this does not obsolete doing at least one compile as you suggested, but is a huge improvement especially for CI or local integration tests.

You would now do:
```bash
$ serverless webpack
$ serverless invoke local --function <function-name-1>
$ serverless invoke local --function <function-name-2>
```

However, I'm still thinking about the disabling of the compile at all, but out from the plugin I did not find a viable solution as it overrides the packaging and execution to point to the compiled code.
